xonsh/environ.py

```diff
@@ -361,6 +364,8 @@ def xonshconfig(env):
         'at the prompt. This may be parameterized in the same way as '
         'the $PROMPT variable. Currently, this is only available in the '
         'prompt-toolkit shell.'),
+    'DYNAMIC_CWD_WIDTH': VarDocs('Target length in number of characters '
+        'for the {dynamic_cwd} prompt var'),
```

Comment:
This should explain how the parsing works with `%`, `inf`, etc.  Also, have you tried running `xonfig` with this branch? I am concerned that the braces `{}` might break it.